precompiles/dapps-staking/src/lib.rs

```diff
@@ -453,6 +480,18 @@ where
             Action::WithdrawUnbounded => Self::withdraw_unbonded(gasometer, context)?,
             Action::ClaimDapp => Self::claim_dapp(input, gasometer, context)?,
             Action::ClaimStaker => Self::claim_staker(input, gasometer, context)?,
+            Action::FreeBalanceRewardDestination => Self::set_reward_destination(
+                RewardDestination::FreeBalance,
+                input,
+                gasometer,
+                context,
+            )?,
+            Action::StakeBalanceRewardDestination => Self::set_reward_destination(
+                RewardDestination::StakeBalance,
+                input,
+                gasometer,
+                context,
+            )?,
```

Comment:
Why not use input parameter instead of 2 methods?

Reply:
I will change it but Solidity enums are primitive (represented by u8) so if we introduce more complex enum type in the future, we'll need to add a different method.